lib/ace/lib/dom.js

```diff
@@ -152,17 +152,22 @@ exports.setCssClass = function(node, className, include) {
 exports.hasCssString = function(id, doc) {
     var index = 0, sheets;
     doc = doc || document;
-    if ((sheets = doc.getElementsByTagName("style"))) {
+    if ((sheets = doc.querySelectorAll("style"))) {
```

Comment:
did you add this to traverse shadow trees? FYI it won't penetrate shadow DOM so you would have to add a little more logic (if you hit something with a shadow root, recursively call this on that)

Reply:
this does not need to penetrate shadow DOM, since it only needs to find style nodes that can change styling of current document-fragment/document